tests/features/async-components.spec.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  defineAsyncComponent,
+  defineComponent,
+  h,
+  ComponentPublicInstance
+} from 'vue'
+import flushPromises from 'flush-promises'
+
+import { mount } from '../../src'
+
+// AsyncComponents are documented here: https://github.com/vuejs/rfcs/blob/async-component/active-rfcs/0026-async-component-api.md
+
+describe('defineAsyncComponent', () => {
+  it('works with the basic usage', async () => {
+    const AsyncHello = defineAsyncComponent(() =>
+      import('../components/Hello.vue')
+    )
+    const Comp = defineComponent({
+      render() {
+        return h('div', [h(AsyncHello)])
+      }
+    })
+
+    const wrapper = mount(Comp)
+    await flushPromises()
+    expect(wrapper.html()).toContain('Hello world')
+  })
+
+  it('works with options usage', async (done) => {
+    const Async = defineAsyncComponent({
+      loader: () =>
+        new Promise<any>((res) => {
+          setTimeout(() => {
+            res({
+              template: '<div>Async Component</div>'
+            })
+          }, 75)
+        }),
+      loadingComponent: {
+        template: '<div>Loading Component</div>'
+      },
+      delay: 10
+    })
+
+    const Comp = defineComponent({
+      render() {
+        return h('div', [h(Async)])
+      }
+    })
+
+    const wrapper = mount(Comp)
+
+    setTimeout(() => {
+      expect(wrapper.html()).toContain('Loading Component')
+    }, 35)
+
+    setTimeout(() => {
+      expect(wrapper.html()).toContain('Async Component')
+      done()
+    }, 100)
+  })
+
+  it('works with vue files', async () => {
```

Comment:
isnt this test the same as the first one?

Reply:
Oops, it is.

I can change this one to test this usage

```
defineAsyncComponent({
  errorComponent: Hello
})
```

where `errorComponent` is a regular `import` at the top of the file
```